Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs

```diff
@@ -0,0 +1,102 @@
+﻿using ReLogic.Content;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Jungle;
+using StarlightRiver.Core.Systems.BarrierSystem;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Terraria.DataStructures;
+using Terraria.GameContent;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Dungeon
+{
+	public class StoneOfTheDrowned : SmartAccessory
+	{
+		internal float slotsUsed; // used so you only summon the amount of slots just gained. Otherwise would populate all of the players remaining slots
+		// i love private terraria methods which require reflection to access
+		public static MethodInfo? playerItemCheckShoot_Info;
+		public static Action<Player, int, Item, int>? playerItemCheckShoot;
+
+		public override string Texture => AssetDirectory.JungleItem + Name;
+
+		public StoneOfTheDrowned() : base("Stone of the Drowned", "+30 barrier\nIncreases your number of max minions by 2 when you have no barrier\nAutomatically summons two slots worth of minions if there is a valid weapon in your inventory") { }
+
+		public override void Load()
+		{
+			playerItemCheckShoot_Info = typeof(Player).GetMethod("ItemCheck_Shoot", BindingFlags.NonPublic | BindingFlags.Instance);
+			//Here we cache this method for performance
+			playerItemCheckShoot = (Action<Player, int, Item, int>)Delegate.CreateDelegate(
+				typeof(Action<Player, int, Item, int>), playerItemCheckShoot_Info);
+
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.value = Item.sellPrice(gold: 2);
+			Item.rare = ItemRarityID.Orange;
+		}
+
+		public override void SafeUpdateEquip(Player Player)
+		{
+			Player.GetModPlayer<BarrierPlayer>().maxBarrier += 30;
+			if (Player.GetModPlayer<BarrierPlayer>().barrier <= 0)
+			{
+				Player.maxMinions += 2;
+
+				if (Player.slotsMinions < Player.maxMinions && slotsUsed < 2)
```

Comment:
Should we seperate out a helper method for finding a valid summon item?

Reply:
no its specific for the item

Reply:
or maybe

Reply:
the spawning of the minions would also need to be a part of the helper method, which i dont think is ideal and very niche